Source/Plugin.BLE/Android/BroadcastReceivers/BondStatusBroadcastReceiver.cs

```diff
@@ -12,26 +12,31 @@ public class BondStatusBroadcastReceiver : BroadcastReceiver
     {
         public event EventHandler<DeviceBondStateChangedEventArgs> BondStateChanged;
 
-        public override void OnReceive(Context context, Intent intent)
+        Adapter BroadCastAdapter;
+
+        public BondStatusBroadcastReceiver(Adapter adapter)
         {
-            var bondState = (Bond)intent.GetIntExtra(BluetoothDevice.ExtraBondState, (int)Bond.None);
-            //ToDo
-            var device = new Device(null, (BluetoothDevice)intent.GetParcelableExtra(BluetoothDevice.ExtraDevice), null);
-            Console.WriteLine(bondState.ToString());
+            BroadCastAdapter = adapter;
+        }
 
+        public override void OnReceive(Context context, Intent intent)
+        {
             if (BondStateChanged == null) return;
 
+            var bondState = (global::Android.Bluetooth.Bond)intent.GetIntExtra(BluetoothDevice.ExtraBondState, (int)global::Android.Bluetooth.Bond.None);
+            var bluetoothDevice = (BluetoothDevice)intent.GetParcelableExtra(BluetoothDevice.ExtraDevice);
+            var device = new Device(BroadCastAdapter, bluetoothDevice, null, 0);
             switch (bondState)
             {
-                case Bond.None:
+                case global::Android.Bluetooth.Bond.None:
```

Comment:
Any reason you introduce all these `global::Android.Bluetooth` references here (although there is a corresponding `using` statement above)? IMHO they makes the code unnecessarily bulky and less readable ...

Reply:
Simplified the code.

Reply:
Nice, thanks 👍